projects/addon-mobile/components/mobile-calendar/mobile-calendar.component.ts

```diff
@@ -247,6 +261,15 @@ export class TuiMobileCalendarComponent {
         return this.value.from.year;
     }
 
+    private idleCheckViewportSize() {
+        this.yearsScrollRef?.scrolledIndexChange
+            .pipe(delay(200), take(1), takeUntil(this.destroy$))
```

Comment:
Instead of another subscription, let's just add filter before `take(1)` of initial subscription that would determine if it's the moment we were waiting for.

Reply:
fixed